### DISCUSSION TITLE
[V5] Validating Uploaded Files

### DISCUSSION THREAD

**Discussion by Pratikkshirsagar** [adonisjs/core#1771] [General]
Hi,
I'm trying to validate my file input in adonis 5 
```ts 
const file = request.file('file', {
      size: '2mb',
      extnames: ['jpg', 'png', 'jpeg', 'pdf'],
    });
```
but it is not validating the file input please help

**Comment by thetutlage** [comment]
Can you please be specific about what you mean by `but it is not validating the file input`?

What you expect to happen and what is not happening?

**Reply by Pratikkshirsagar** [reply]
Sorry for not being specific ```I mean to say that file size and file type are not validating.```
when I send the csv file it should not be accept because I have not mention it in the ```extnames array``` also the size should be restricted to 2mb and not more than that.

**Reply by thetutlage** [reply]
How do you verify whether it is accepting the file or not? Did you try checking the errors by logging `file.errors`?

**Reply by Pratikkshirsagar** [reply]
I was not explicitly checking ```files.errors```  which is the array of error log, Thank you.

**Reply by thetutlage** [reply]
I recommend reading the docs here https://preview.adonisjs.com/guides/http/file-uploads#validating-uploaded-files
